pkg/operator/resource/resourceapply/volumesnapshotclass.go

```diff
@@ -0,0 +1,96 @@
+package resourceapply
+
+import (
+	"context"
+
+	"github.com/imdario/mergo"
+	"k8s.io/klog/v2"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const (
+	VolumeSnapshotClassGroup    = "snapshot.storage.k8s.io"
+	VolumeSnapshotClassVersion  = "v1"
+	VolumeSnapshotClassResource = "volumesnapshotclasses"
+)
+
+var volumeSnapshotClassResourceGVR schema.GroupVersionResource = schema.GroupVersionResource{
+	Group:    VolumeSnapshotClassGroup,
+	Version:  VolumeSnapshotClassVersion,
+	Resource: VolumeSnapshotClassResource,
+}
+
+func ensureGenericVolumeSnapshotClass(required, existing *unstructured.Unstructured) (*unstructured.Unstructured, bool, error) {
+	requiredParameters, _, err := unstructured.NestedMap(required.UnstructuredContent(), "parameters")
+	if err != nil {
+		return nil, false, err
+	}
+	existingParameters, _, err := unstructured.NestedMap(existing.UnstructuredContent(), "parameters")
+	if err != nil {
+		return nil, false, err
+	}
+
+	if err := mergo.Merge(&existingParameters, &requiredParameters); err != nil {
+		return nil, false, err
+	}
+
+	if equality.Semantic.DeepEqual(existingParameters, requiredParameters) {
+		return existing, false, nil
+	}
+
+	existingCopy := existing.DeepCopy()
+	if err := unstructured.SetNestedMap(existingCopy.UnstructuredContent(), existingParameters, "parameters"); err != nil {
```

Comment:
This was the typo we found in the spec version of this.  You want the requiredParameters here, right?